pkg/sql/repair.go

```diff
@@ -780,17 +781,22 @@ func (p *planner) ForceDeleteTableData(ctx context.Context, descID int64) error
 	}
 
 	prefix := p.extendedEvalCtx.Codec.TablePrefix(uint32(id))
-	tableSpans := roachpb.Span{Key: prefix, EndKey: prefix.PrefixEnd()}
+	tableSpan := roachpb.Span{Key: prefix, EndKey: prefix.PrefixEnd()}
+	requestHeader := roachpb.RequestHeader{
+		Key: tableSpan.Key, EndKey: tableSpan.EndKey,
+	}
 	b := &kv.Batch{}
-	b.AddRawRequest(&roachpb.ClearRangeRequest{
-		RequestHeader: roachpb.RequestHeader{
-			Key:    tableSpans.Key,
-			EndKey: tableSpans.EndKey,
-		},
-	})
-
-	err = p.txn.DB().Run(ctx, b)
-	if err != nil {
+	if p.execCfg.Settings.Version.IsActive(ctx, clusterversion.UseDelRangeInGCJob) {
+		b.AddRawRequest(&roachpb.DeleteRangeRequest{
+			RequestHeader:     requestHeader,
+			UseRangeTombstone: true,
+		})
```

Comment:
Do we need to worry about unbounded parallelization here? Also, is it worth adding in admission control headers?

Reply:
We didn't in the `ClearRange` code which preceded this. I think we have all the same fast-paths now, no?

Reply:
I guess the argument is that the user has decided to take this dire action. I don't know. I honestly have never used this thing. It came out of a post-mortem. I don't care about it really.

Reply:
Hm ok, probably fine then. Seems like the sort of thing you'd pull in an emergency.